cmd/dockle/main.go

```diff
@@ -14,7 +14,11 @@ var (
 	version = "dev"
 )
 
-func main() {
+/*
+NewDockleCli Factory for Dockle CLI creation.
+An Enabler for programmatic usage of Dockle
+*/
+func NewDockleCli() *cli.App {
```

Comment:
It looks cool to name this function `NewApp()`. 
You can use it as `app := dockle.NewApp()`.